controllers/datadogagent/utils.go

```diff
@@ -501,6 +501,9 @@ func getEnvVarsForProcessAgent(dda *datadoghqv1alpha1.DatadogAgent) ([]corev1.En
 			return nil, err
 		}
 		envVars = append(envVars, envs...)
+
+		// The process agent retrieves the cluster id from the Cluster Agent
+		envVars = append(envVars, envForClusterAgentConnection(dda)...)
```

Comment:
Inside `getEnvVarsForAgent`, the appending of `envForClusterAgentConnection(dda)…` is gated by a `if spec.ClusterAgent != nil`.
Shouldn’t we also add the DCA env vars to the `process-agent` only if the DCA is enabled ?

Reply:
that's why I've put a `if dda.Spec.ClusterAgent != nil` inside `envForClusterAgentConnection` as a safegard

Reply:
i.e `envForClusterAgentConnection` returns an empty list if the DCA is disabled

Reply:
Indeed ! 👍🏻